operator/apis/loki/v1/v1.go

```diff
@@ -28,6 +28,10 @@ const (
 	StorageSchemaUpdateBuffer = time.Hour * 2
 )
 
+const (
+	AnnotationCustomTopology = "loki.grafana.com/custom-tenant-topology"
```

Comment:
I think I would prefer this annotation to be called "what it does" (disable tenant-ID validation) rather than "what it is for" (enabling custom tenant-IDs):

```suggestion
	AnnotationDisableTenantValidation = "loki.grafana.com/disable-tenant-validation"
```